lib/phraseapp_updater/locale_file.rb

```diff
@@ -1,18 +1,28 @@
-require 'multi_json'
-require 'oj'
-
-# We're working with pure JSON, not
-# serialized Ruby objects
-Oj.default_options = {mode: :strict}
+require "phraseapp_updater/locale_file/json_file"
+require "phraseapp_updater/locale_file/yaml_file"
 
 class PhraseAppUpdater
   class LocaleFile
     attr_reader :name, :content, :parsed_content
 
+    class BadFileTypeError < StandardError ; end
+
     def self.from_hash(name, hash)
-      new(name, MultiJson.dump(hash))
+      raise RuntimeError.new("Must be implemented in a subclass.")
+    end
+
+    def self.class_for_file_format(type)
+      case type.downcase
```

Comment:
If you do want to be able to print out supported types in error messages, you might want to change this from a `case` statement to a string -> Class hash.